src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosStatus, setTodosStatus] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setHasLoadingError(true);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const filterTodo = todos.filter((todo) => {
+    switch (todosStatus) {
+      case 'active':
+        return !todo.completed && todo.title.toLowerCase().includes(query.toLowerCase());
+      case 'completed':
+        return todo.completed && todo.title.toLowerCase().includes(query.toLowerCase());
+      default:
+        return todo.title.toLowerCase().includes(query.toLowerCase());
+    }
+  });
```

Comment:
would be better if you could know about errors if you have an error.
you can "throw new Error(error)" or console.error(error) 

```suggestion
  const filterTodo = todos.filter((todo) => {
    switch (todosStatus) {
      case 'active':
        return !todo.completed && todo.title.toLowerCase().includes(query.toLowerCase());
      case 'completed':
        return todo.completed && todo.title.toLowerCase().includes(query.toLowerCase());
      default:
        return todo.title.toLowerCase().includes(query.toLowerCase());
    }
  });
  
  const loadTodos = useCalback(async  () => {
      try {
        const todosFromServer = await getTodos();

        setTodos(todosFromServer);
      } catch (error) {
        setHasLoadingError(true);
        throw new Error(e);
      }
    }, []);

  useEffect(() => {
    loadTodos();
  }, []);
```